wavepool/views.py

```diff
@@ -14,9 +14,16 @@ def front_page(request):
             archive: the rest of the newsposts, sorted by most recent
     """
     template = loader.get_template('wavepool/frontpage.html')
-    cover_story = NewsPost.objects.all().order_by('?').first()
-    top_stories = NewsPost.objects.all().order_by('?')[:3]
-    other_stories = NewsPost.objects.all().order_by('?')
+    # this is to place the cover story within its proper location. However, is cover story MUST be unique. Alternatively .filter(is_cover_story=True).get()
+    try:
+        cover_story = NewsPost.objects.get(is_cover_story=True)
```

Comment:
This is only for non-ideal cases, the intended behavior is that there will ALWAYS be one cover and ONLY one, in any other scenario we can still display a random story, though in hindsight it makes sense to display the most recent story instead. This is much better than using ? which is not ideal performance wise for building this list

Reply:
** building this query set